### ISSUE TITLE
bug in decomposeFlowDecomposition

### ISSUE THREAD

**Issue by videlec** [flatsurf/flatsurf#140]
The code snippet below shows a Python example where `decomposeFlowDecomposition` does not terminate
```
from pyeantic import eantic
from pyflatsurf import flatsurf, Surface

K = eantic.renf_class.make("l^2 - l - 8", "l", "3.37 +/- 0.01")
l = K.gen()
R = eantic.renf_elem
R2 = flatsurf.Vector['eantic::renf_elem_class']
z = K.zero()
o = K.one()
vectors = [
    R2(4*l+1, 4*l),
    R2(-4*l, z),
    R2(-o, -4*l),
    R2(-3*o, -8*o),
    R2(-4*l+3, 8*o),
    R2(4*l, z),
    R2(4*l-19, -8*o),
    R2(-4*l+15, z),
    R2(4*o, 8*o),
    R2(-4*o, -8*o),
    R2(-3*o, -8*o),
    R2(-o, z)
    ]
vertices = [[1, -3, -6, 11, -12, 9, -8, 7, -9, 4, -5, 6], [-1, 3, -2, -4, 12, 10, 8, -7, -10, -11, 5, 2]]
surface = Surface(vertices, vectors)
v = R2(4*l-11, 4*l+8)
decomposition = flatsurf.makeFlowDecomposition(surface, v)
flatsurf.decomposeFlowDecomposition(decomposition, -1)
```
If in the last line we replaced `-1` with any non-negative number it works just fine and finds the decomposition which is made of two components without periodic trajectories.

Similar example at #136 with the (4,5,5) triangle.

**Comment by saraedum** [comment]
Is this still an issue? If so, could you print the surface so I can add it to the test cases?

**Comment by videlec** [comment]
Still an issue. However, it is maybe not possible to decide the nature of the direction `((a-11/4 ~ 0.62228132), (a+2 ~ 5.3722813))`. Since `H(1,1)` is in small dimension, I thought that it should not be of any trouble.

**Comment by saraedum** [comment]
@videlec So what should we do about this one?

**Comment by videlec** [comment]
More precision at https://github.com/flatsurf/flatsurf/issues/202

**Comment by videlec** [comment]
I think the stopping condition for testing minimal components is wrong. Here it looks like we know its minimal but the decomposition code still want to apply Zorich induction.

**Comment by saraedum** [comment]
This might be related to #203.

**Comment by videlec** [comment]
fixed in release 1.1.2
